flink-streaming-java/src/main/java/org/apache/flink/streaming/util/typeutils/FieldAccessorFactory.java

```diff
@@ -30,14 +30,23 @@
 import org.apache.flink.api.java.typeutils.TupleTypeInfo;
 import org.apache.flink.api.java.typeutils.TupleTypeInfoBase;
 
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
 import java.io.Serializable;
+import java.util.Optional;
 import java.util.regex.Matcher;
 import java.util.regex.Pattern;
 
 /** Static factories for the {@link FieldAccessor} utilities. */
 @Internal
 public class FieldAccessorFactory implements Serializable {
 
+    private static final Logger LOG = LoggerFactory.getLogger(FieldAccessorFactory.class);
+
+    private static final Optional<ScalaProductFieldAccessorFactory>
```

Comment:
IIRC we use nullable for field. Also as we check `isPresent` manually, use `Optional` doesn't help a lot.

Reply:
I'll change it.